src/azure-cli/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -255,78 +391,185 @@ class CLIJsonWebKeyOperation(str, Enum):
+        c.argument('sd_exchange_key', help='Local file path to store the exported key.')
+
+    with self.argument_context('keyvault security-domain upload') as c:
+        c.argument('sd_file', help='This file contains security domain encrypted using SD Exchange file downloaded '
+                                   'in security-domain init-recovery command.')
+        c.argument('sd_exchange_key', help='The exchange key for security domain.')
+        c.argument('sd_wrapping_keys', nargs='*',
+                   help='Space-separated file paths to PEM files containing private keys.')
+        c.argument('passwords', nargs='*', help='Space-separated password list for --sd-wrapping-keys. '
+                                                'CLI will match them in order. Can be omitted if your keys are without '
+                                                'password protection.')
+
+    with self.argument_context('keyvault security-domain download') as c:
+        c.argument('sd_wrapping_keys', nargs='*',
+                   help='Space-separated file paths to PEM files containing public keys.')
+        c.argument('security_domain_file',
+                   help='Path to a file where the JSON blob returned by this command is stored.')
+        c.argument('sd_quorum', type=int, help='The minimum number of shares required to decrypt the security domain '
+                                               'for recovery.')
+
+    with self.argument_context('keyvault security-domain wait') as c:
+        c.argument('hsm_base_url', hsm_name_type, help='Name of the HSM. Can be omitted if --id is specified.',
+                   required=False)
+        c.argument('identifier', options_list=['--id'], validator=validate_vault_or_hsm, help='Id of the HSM.')
+        c.argument('resource_group_name', options_list=['--resource-group', '-g'],
+                   help='Proceed only if HSM belongs to the specified resource group.')
+    # endregion
+
+    # region keyvault backup/restore
+    for item in ['backup', 'restore']:
+        for scope in ['start', 'status']:
+            with self.argument_context('keyvault {} {}'.format(item, scope), arg_group='HSM Id') as c:
+                c.argument('vault_base_url', hsm_name_type, required=False,
+                           help='Name of the HSM. Can be omitted if --id is specified.')
+                c.extra('identifier', options_list=['--id'], validator=validate_vault_or_hsm, help='Id of the HSM.')
+                c.ignore('cls')
+
+    with self.argument_context('keyvault backup start', arg_group='Storage Id') as c:
+        c.argument('storage_resource_uri', required=False, validator=process_storage_uri,
+                   help='Azure Blob storage container Uri. If specified all other \'Storage Id\' arguments '
+                        'should be omitted')
+        c.extra('storage_account_name', help='Name of Azure Storage Account.')
```

Comment:
with storage blob container uri, storage accoutn name and blob container name could be parsed from it. so either storage_resource_uri or (storage_accoutn_name+blob_container_name) right?